packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -609,11 +609,60 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
   /// are required to wait for the returned future to complete before calling
   /// this method again. Attempts to do otherwise will result in a
   /// [TestFailure] error being thrown.
+  ///
+  /// If your test code depends on the result of a task that did not complete,
+  /// the most common symptom would be a hanging test as FakeAsync environment
+  /// is unable to resolve a future that was created in `runAsync`. If you
+  /// observe such behavior or flakiness, you have a number of options:
+  ///
+  /// - Consider restructuring your code so you do not need `runAsync`. This is
+  ///   the optimal solution as widget tests are designed to run in Fake async
+  ///   environment.
+  ///
+  /// - Expose a [Future] in your application code that signals the readiness of
+  ///   your widget tree, then await that future inside `callback`.
+  ///
+  /// - If either option is not feasible, await [resolveFutures] inside
+  ///   `callback`.
   Future<T> runAsync<T>(
     Future<T> callback(), {
     Duration additionalTime = const Duration(milliseconds: 1000),
   }) => binding.runAsync<T>(callback, additionalTime: additionalTime);
 
+  /// Waits for a stable state within a single `runAsync` execution.
+  ///
+  /// Note that using this method is *not* recommended. This method should only
+  /// be used when there are unawait-able side effects that your test depends
+  /// on.
+  ///
+  /// [runAsync] method keeps track of microtasks and timers created during
+  /// `callback`. [resolveFutures] creates a periodic timer that checks the
+  /// status of these timers and returns once there are no active timers *and*
+  /// no new microtasks have been created for a certain period of time. In other
+  /// words, it attempts to observe a certain length of "idle" period specified
+  /// by `expectedIdlePeriod`.
+  ///
+  /// Microtasks might still be generated after this idle period, therefore the
+  /// completion of this method does not mean the application is in a stable
+  /// state. It also means each test that awaits this Future extends its runtime
+  /// by this idle period since we have to wait *at least* that length of time
+  /// to observe the idleness. This is why [runAsync] does not await on
+  /// `resolveFutures` by default.
+  ///
+  /// Setting `expectedIdlePeriod` is a tradeoff between how much longer your
+  /// test will take and how confident we are of the idle state. If you set it
+  /// too long, you will waste a lot of test time. If you set it too short, your
+  /// tests might be flaky. See other recommended approaches for dealing with
```

Comment:
> tests might be flaky

Adding another option to make your tests flaky doesn't seem like a good idea.

Reply:
it doesn't matter how long you set it, it'll always be flaky